### PR TITLE
Logger levels

### PR DESCRIPTION
**Pull request by hemalathagajendran**
## Pull request checklist

```release-notes

```

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature Enhancement
- [ ] Test Automation 
- [ ] Code Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## What is the current behavior?

-

## What is the new behavior?

-

## How risky is this change?
- [ ] Small, isolated change
- [ ] Medium, requires regression testing
- [ ] Large, requires functional and regression testing

### CODE REVIEW

**Comment by hemalathagajendran** [comment]
As part of this PR different logger levels are maintained. Six log levels (DEBUGPLUS, DEBUG,INFO,WARNING,ERROR,FATAL) are defined in CSI. Log level are defined in config map and used as environment variable in driver code. "LOGLEVEL" is the env variable maintained in config map. Default level would be "INFO". Logs will be written to both files and stderr. Log file path would be under /host/var/log directory in pod. Depending on the severity  log events at or above the described severity are logged to standard error. However irrespective of level all logs (except debug) would be written to log files.

File: operator/controllers/syncer/csi_syncer.go [hunk 1]
```diff
@@ -559,7 +559,7 @@ func (s *csiControllerSyncer) ensureProvisionerContainersSpec() []corev1.Contain
 		s.getSidecarImage(config.CSIProvisioner),
 		// TODO: make timeout configurable
 		[]string{"--csi-address=$(ADDRESS)", "--timeout=3m", "--worker-threads=10",
-			"--extra-create-metadata", "--v=5", "--default-fstype=gpfs",
+			"--extra-create-metadata", "--default-fstype=gpfs",
```
Comment (Jainbrt) [inline operator/controllers/syncer/csi_syncer.go:562]: not sure, if we are still managing the sidecar log level at present from configmap?

**Comment by Jainbrt** [comment]
CSI logs related to version/commit id, should be print for every log level. 

```
I0103 07:42:46.999564       1 main.go:49] [a1cbccca-8f08-4c1f-bd89-1c8757878fc9] Version Info: commit ()
I0103 07:42:46.999973       1 gpfs.go:120] [a1cbccca-8f08-4c1f-bd89-1c8757878fc9] gpfs GetScaleDriver
I0103 07:42:46.999989       1 gpfs.go:196] [a1cbccca-8f08-4c1f-bd89-1c8757878fc9] gpfs SetupScaleDriver. name: spectrumscale.csi.ibm.com, version: 2.8.0, nodeID: worker2.csi290.cp.fyre.com
```

File: operator/controllers/syncer/csi_syncer.go (hunk 1 above)
Comment (hemalathagajendran) [reply operator/controllers/syncer/csi_syncer.go]: We are not managing sidecar log level from config map... It is taken from the pod spec. I have removed --v=5 from podspec for driver container alone as it overrides the value which is read as env from configmap

**Comment by hemalathagajendran** [comment]
> CSI logs related to version/commit id, should be print for every log level.
> 
> ```
> I0103 07:42:46.999564       1 main.go:49] [a1cbccca-8f08-4c1f-bd89-1c8757878fc9] Version Info: commit ()
> I0103 07:42:46.999973       1 gpfs.go:120] [a1cbccca-8f08-4c1f-bd89-1c8757878fc9] gpfs GetScaleDriver
> I0103 07:42:46.999989       1 gpfs.go:196] [a1cbccca-8f08-4c1f-bd89-1c8757878fc9] gpfs SetupScaleDriver. name: spectrumscale.csi.ibm.com, version: 2.8.0, nodeID: worker2.csi290.cp.fyre.com
> ```

By default this information would be there for INFO,DEBUG and trace level. However we could not see this info in ERROR and FATAL level. We could not make it as fatal also as program would exit at fatal level.

**Review by Jainbrt — APPROVED** [review]

**Pull request merged**